File: eunjin/Problem/탐색/부품찾기.cpp
//이분탐색
#include <iostream>
#include <vector>
#include <algorithm>

int N, M;
std::vector<int> arr, target;

int binary_search(std::vector<int> arr, int start, int end, int target)
{
	int middle;

	while (start <= end)
	{
		middle = (start + end) / 2;
		if (arr[middle] == target)
			return middle;
		else if (arr[middle] > target)
			end = middle - 1;
		else
			start = middle + 1;
	}
	return -1;
}

int main()
{
	int tmp;
	std::cin >> N;
	for (int i = 0; i < N; i++)
	{
		std::cin >> tmp;
		arr.push_back(tmp);
	}

	std::sort(arr.begin(), arr.end());

	std::cin >> M;

	for (int i = 0; i < M; i++)
	{
		std::cin >> tmp;
		target.push_back(tmp);
	}

	for (int i = 0; i < M; i++)
	{
		//해당 부품이 있는지 확인 
		if (binary_search(arr, 0, N - 1, target[i]) != -1) //arr 배열 최대를 검사해야 한다
			std::cout << "yes \n";
		else
			std::cout << "no \n";
	}

	return 0;
}

File: heejeong/BinarySearch/부품찾기.py
def binarySearch(nums, target, start, end):

    while start <= end:
        mid = (start+end) // 2
        if nums[mid] == target:
            return mid
        elif nums[mid] < target:
            start = mid + 1
        elif nums[mid] > target:
            end = mid - 1
    return None  # 부품에 없는 경우


n = int(input())  # 부품 종류개수
nums = list(map(int, input().split()))
m = int(input())  # 손님이 요청한 부품
order = list(map(int, input().split()))

nums.sort()

for i in order:
    result = binarySearch(nums, i, 0, n-1)
    if result != None:
        print('yes', end=' ')
    else:
        print('no', end=' ')
